backend/src/data/repository/user.ts

```diff
@@ -44,6 +32,43 @@ export class UserRepository {
     return data;
   }
 
+  async getUsers(): Promise<DTO.IUserDoc[] | null> {
+    const data = await User.find(
+      {},
+      { firstName: 1, lastName: 1, photo: 1, birthday: 1, country: 1, speak: 1, learn: 1 }
+    ).limit(5);
+    return data;
+  }
+  // Search
+
+  async getUsersByParams(params: DTO.FilterRequest): Promise<DTO.IUserDoc[] | null> {
+    const { name, lowAge, highAge, sex, country, language, level } = params;
+    const data = await User.find({
+      $and: [
+        {
+          $expr: {
+            $regexMatch: {
+              input: { $concat: ['$firstName', ' ', '$lastName'] },
+              regex: name,
```

Comment:
It is not recommended to pass params as a regex. Because search of `(abc` will lead to invalid regex error, also regex are slow.
It is better to search when `firstname lastname` includes `name` and case-insensitive.